### ISSUE TITLE
fcma: `random` argument in `preprocessing.prepare_fcma_data` is not clear

### ISSUE THREAD

**Issue by moopet17** [brainiak/brainiak#363]
I ran the following code:

`permuted_raw_data, _, permuted_labels = prepare_fcma_data(images, epoch_list, mask, random=2)
print(permuted_labels)`

which gives me this output: 
[0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1]

However, I get the exact same output, when I run the same code again or when I set `random=0`, `random=1`, `random=2`, or when I leave out the random argument altoghether. From what I understand, this should not happen, as the lables should be shuffled? Do I get the documentation wrong or might there be a bug?

Also, I the end, my fcma results give me the same classification performance, regardless of what I set the random argument to.

**Comment by yidawang** [comment]
Setting the `random` argument to 1 or 2 shuffles the data not the labels. http://brainiak.org/docs/brainiak.fcma.html#brainiak.fcma.preprocessing.RandomType

What is your dataset? What is the classification results you got from 'random=0`?

**Comment by mihaic** [comment]
@yidawang, what do you think about changing the docstring to say "images" instead of (or in addition to) "data", to indicate as clearly as possible what is shuffled?

**Comment by yidawang** [comment]
@mihaic sounds good, I can do it later today.

**Comment by mihaic** [comment]
@moopet17, hopefully PR #365 makes it clear that the images are what gets shuffled.

If you are still worried about the classification performance not changing, please provided some data to reproduce what you are seeing.

**Comment by moopet17** [comment]
@yidawang: Thank you.
@mihaic: Yes, that makes it clear. It would be great if the documentation would also be clarified here: http://brainiak.org/docs/brainiak.fcma.html?highlight=fcma%20preprocessing%20prepare_fcma_data#brainiak.fcma.preprocessing.prepare_fcma_data This is the first place to look if one tries to understand what a BrainIAK function does.

**Comment by mihaic** [comment]
At this moment, we only have HTML documentation for BrainIAK releases. I have just created issue #367 to add a version of the HTML docs corresponding to the master branch of BrainIAK.

How about the classification performance? Is it unexpected? If not, please close this issue.

**Comment by moopet17** [comment]
This is the dataset: https://www.dropbox.com/s/11adrjdkt0w1tr3/face_scene.zip?dl=0
from this study: https://doi.org/10.1016/j.jneumeth.2015.05.012

Yes, the classification performance is unexpected, as described in my initial post. Permuting the data should not result in the same performance as not permuting it.
This is my code:

`data_dir = './face_scene/'
suffix = 'bet.nii.gz'
images = io.load_images_from_dir(data_dir, suffix)

mask_file =  data_dir + 'mask.nii.gz'
mask = io.load_boolean_mask(mask_file)

epoch_file = data_dir + 'fs_epoch_labels.npy'
epoch_list = io.load_labels(epoch_file)

permuted_raw_data, _, permuted_labels = prepare_fcma_data(images, epoch_list, mask, random=2) `

**Comment by moopet17** [comment]
Ok the real issue seems to be that TO ME, the documentation suggested that `random=1` equals `random=RandomType.REPRODUCIBLE` and so on. Turns out that is not the case. Setting the random argument to an integer does not do anything and always defaults to `RandomType.NORANDOM`. So it seems to me that this beahavior should be fixed or the documentation should be clearer on that.

**Comment by mihaic** [comment]
I agree, the documentation can be interpreted as allowing `random=1`. We can improve it.

Otherwise, can you please confirm that the classification performance changes as expected when using `UNREPRODUCIBLE`?
